### PR TITLE
Expand connection timeout for `presignedPostPolicy` functional test

### LINKED ISSUE #716: Azure Gateway: presignedPostPolicy functional test fails
presignedPostPolicy test failed on Azure gateway.
```
{"name":"minio-java","function":"presignedPostPolicy(PostPolicy policy)","duration":10028,"status":"FAIL","error":"java.net.SocketTimeoutException: timeout >>> [okio.Okio$4.newTimeoutException(Okio.java:227), okio.AsyncTimeout.exit(AsyncTimeout.java:284), okio.AsyncTimeout$2.read(AsyncTimeout.java:240), okio.RealBufferedSource.indexOf(RealBufferedSource.java:344), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:216), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:210), okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:75), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185), okhttp3.RealCall.execute(RealCall.java:69), FunctionalTest.presignedPostPolicy_test(FunctionalTest.java:1803), FunctionalTest.runTests(FunctionalTest.java:2735), FunctionalTest.main(FunctionalTest.java:2843)]"}
```

### PR DESCRIPTION
**Pull request by Praveenrajmani**
There was a socket timeout exception over azure gateway for `presignedPostPolicy` test case. Hence, increased the connection timeout to 20 secs ( from 10secs - which is default )
Fixes #716

### CODE REVIEW

File: functional/FunctionalTest.java [hunk 1]
```diff
@@ -1897,7 +1898,12 @@ public static void presignedPostPolicy_test() throws Exception {
       String urlString = client.getObjectUrl(bucketName, "");
       Request request = requestBuilder.url(urlString).post(multipartBuilder.build()).build();
       OkHttpClient transport = new OkHttpClient();
-      Response response = transport.newCall(request).execute();
+      Response response = transport.newBuilder()
+          .connectTimeout(20, TimeUnit.SECONDS)
+          .writeTimeout(20, TimeUnit.SECONDS)
+          .readTimeout(20, TimeUnit.SECONDS)
+          .build()
```
Comment (balamurugana) [inline functional/FunctionalTest.java]: Assign to `transport` then `transport.newCall()`.  This is easier for debug.

**Comment by kannappanr** [comment]
There was a socket timeout exception in Travis, https://travis-ci.org/minio/minio-java/jobs/446112546 looks like https://github.com/minio/minio-java/blob/master/functional/FunctionalTest.java#L202 threw the exception. So, you might have to increase the timeout in the case of `https://github.com/minio/minio-java/blob/master/functional/FunctionalTest.java#L180` and `https://github.com/minio/minio-java/blob/master/functional/FunctionalTest.java#L217`

**Comment by Praveenrajmani** [comment]
This PR is ready for review @nitisht @balamurugana

**Comment by Praveenrajmani** [comment]
looks like oraclejdk10 is deprecated . I guess we can remove it

**Comment by nitisht** [comment]
Travis fails for `oraclejdk10` as it is deprecated. We can remove `oraclejdk10` tests in a separate PR. Ignoring the failure for now.

**Pull request merged**